app/assets/javascripts/actions/story.js

```diff
@@ -227,13 +221,20 @@ export const dragDropStory =
 
     const newStory = Story.addNewAttributes(story, newAttributes);
 
+    const storiesWithUpdatedPositions = Story.sortOptimistically(
+      stories,
+      newStory,
+      from
+    );
+
     try {
-      dispatch(optimisticallyUpdate(newStory, from));
+      // Optimistic Update:
+      dispatch(sortStories(storiesWithUpdatedPositions, from));
```

Comment:
I was calling an action called "optimisticallyUpdate," which was updating incorrectly the stories position (was fixing the position only of the moved story, not the others). To solve this, I added a function called "sortOptimistically" that corrects the position of all stories. This returns an array identical to the one the server returns. This way, a single action ("sortStories") can perform optimistic updating (fonrt-end data) and then update the server data since the data should be the same.